src/pages/Cart/Cart.js

```diff
@@ -0,0 +1,59 @@
+import React, { useLayoutEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCartItems } from '../../axios/cart/requests';
+import CartItems from '../../components/CartItems/CartItems';
+import CartTotal from '../../components/CartTotal/CartTotal';
+import AddToCartForm from '../../components/Forms/AddToCartForm/AddToCartForm';
+import Section from '../../hoc/Section/Section';
+import { showAlert } from '../../store/actions/alert';
+import {
+  loadCartItems,
+  startLoading,
+  stopLoading,
+} from '../../store/actions/cart';
+import classes from './Cart.module.css';
+
+const Cart = () => {
+  const { items, loading } = useSelector((state) => state.cart);
+  const dispatch = useDispatch();
+
+  useLayoutEffect(() => {
```

Comment:
Probably here you has multiple unnecessary `fetchData` requests due to call after every DOM update.
Please read https://redux.js.org/tutorials/essentials/part-5-async-logic#dispatching-thunks-from-components

Reply:
improved fetching data to use it only when needed https://github.com/maxSedelnikov/react-shopping-app/commit/24984c3c8b52e6eb5707ea4a4edfa8373dc5a402

Reply:
Still not sure that implementation is correct.
You fixed the wrong behaviour, yes! But way how data loading is implemented look like missplaced.

`userLayoutEffect` does not look like a right place to fetch data. I prefered `componentDidMount`

Reply:
useLayoutEffect is the same as componentDidMount but for functional components which is according to the doc
https://prnt.sc/111ez2x